DEVELOPER_GUIDE.md

```diff
@@ -88,12 +88,22 @@ Please follow these formatting guidelines:
 OpenSearch neural-search uses a [Gradle](https://docs.gradle.org/6.6.1/userguide/userguide.html) wrapper for its build.
 Run `gradlew` on Unix systems.
 
-Build OpenSearch neural-search using `gradlew build`
+Build OpenSearch neural-search using `gradlew build`. This command will
+also run Integration Tests and Unit Tests.
```

Comment:
[Question]
build command will run the integ and unit test are we sure?

Reply:
Sorry, accidentally deleted my comment.  I observed this from the behavior Ive seen but just to be sure I ran `./gradlew build | grep "> TASK"`. You can see it does integTest and Test.
```
> Task :generateNotice
> Task :extractKnnJar UP-TO-DATE
> Task :generateEffectiveLombokConfig UP-TO-DATE
> Task :spotlessInternalRegisterDependencies UP-TO-DATE
> Task :spotlessJava
> Task :spotlessJavaApply
> Task :spotlessLicense UP-TO-DATE
> Task :spotlessLicenseApply UP-TO-DATE
> Task :spotlessMisc UP-TO-DATE
> Task :spotlessMiscApply UP-TO-DATE
> Task :spotlessApply
> Task :compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar UP-TO-DATE
> Task :copyPluginPropertiesTemplate UP-TO-DATE
> Task :pluginProperties UP-TO-DATE
> Task :bundlePlugin
> Task :generatePomFileForNebulaPublication
> Task :generatePomFileForPluginZipPublication
> Task :generatePom
> Task :delombok
> Task :javadoc
> Task :sourcesJar
> Task :assemble
> Task :generateTestFixturesEffectiveLombokConfig UP-TO-DATE
> Task :compileTestFixturesJava UP-TO-DATE
> Task :generateTestEffectiveLombokConfig UP-TO-DATE
> Task :compileTestJava UP-TO-DATE
> Task :processTestResources UP-TO-DATE
> Task :testClasses UP-TO-DATE
> Task :processTestFixturesResources NO-SOURCE
> Task :testFixturesClasses UP-TO-DATE
> Task :forbiddenApisResources UP-TO-DATE
> Task :forbiddenApisMain
> Task :testFixturesJar UP-TO-DATE
> Task :forbiddenApisTest
> Task :forbiddenApisTestFixtures
> Task :forbiddenApis
> Task :dependencyLicenses SKIPPED
> Task :filepermissions
> Task :forbiddenPatterns SKIPPED
> Task :jarHell
> Task :licenseHeaders
> Task :loggerUsageCheck SKIPPED
> Task :testingConventions SKIPPED
> Task :thirdPartyAuditResources UP-TO-DATE
> Task :thirdPartyAudit SKIPPED
> Task :validateNebulaPom SKIPPED
> Task :validatePluginZipPom UP-TO-DATE
> Task :validatePom UP-TO-DATE
> Task :precommit
> Task :integTest    ## <----------------------------------------Here
> Task :test              ## <----------------------------------------Here
> Task :jacocoTestReport
> Task :jacocoTestCoverageVerification
> Task :spotlessJavaCheck
> Task :spotlessLicenseCheck UP-TO-DATE
> Task :spotlessMiscCheck UP-TO-DATE
> Task :spotlessCheck
> Task :check
> Task :build
> Task :qa:generatePomFileForNebulaPublication
> Task :qa:generatePom
> Task :qa:generateEffectiveLombokConfig UP-TO-DATE
> Task :qa:compileJava NO-SOURCE
> Task :qa:processResources NO-SOURCE
> Task :qa:classes UP-TO-DATE
> Task :qa:jar UP-TO-DATE
> Task :qa:delombok NO-SOURCE
> Task :qa:javadoc NO-SOURCE
> Task :qa:javadocJar UP-TO-DATE
> Task :qa:sourcesJar UP-TO-DATE
> Task :qa:assemble SKIPPED
> Task :qa:generateTestEffectiveLombokConfig UP-TO-DATE
> Task :qa:compileTestJava NO-SOURCE
> Task :qa:processTestResources NO-SOURCE
> Task :qa:testClasses UP-TO-DATE
> Task :qa:forbiddenApisResources UP-TO-DATE
> Task :qa:forbiddenApisMain NO-SOURCE
> Task :qa:forbiddenApisTest NO-SOURCE
> Task :qa:forbiddenApis UP-TO-DATE
> Task :qa:dependencyLicenses SKIPPED
> Task :qa:filepermissions NO-SOURCE
> Task :qa:forbiddenPatterns SKIPPED
> Task :qa:jarHell
> Task :qa:licenseHeaders NO-SOURCE
> Task :qa:loggerUsageCheck SKIPPED
> Task :qa:testingConventions SKIPPED
> Task :qa:thirdPartyAuditResources UP-TO-DATE
> Task :qa:thirdPartyAudit SKIPPED
> Task :qa:validateNebulaPom SKIPPED
> Task :qa:validatePom UP-TO-DATE
> Task :qa:precommit
> Task :qa:integTest SKIPPED
> Task :qa:spotlessJava
> Task :qa:spotlessJavaCheck
> Task :qa:spotlessLicense UP-TO-DATE
> Task :qa:spotlessLicenseCheck UP-TO-DATE
> Task :qa:spotlessMisc UP-TO-DATE
> Task :qa:spotlessMiscCheck UP-TO-DATE
> Task :qa:spotlessCheck
> Task :qa:test SKIPPED
> Task :qa:check
> Task :qa:build SKIPPED
> Task :qa:restart-upgrade:generatePomFileForNebulaPublication
> Task :qa:restart-upgrade:generatePom
> Task :qa:restart-upgrade:generateEffectiveLombokConfig UP-TO-DATE
> Task :qa:restart-upgrade:compileJava NO-SOURCE
> Task :qa:restart-upgrade:processResources NO-SOURCE
> Task :qa:restart-upgrade:classes UP-TO-DATE
> Task :qa:restart-upgrade:jar UP-TO-DATE
> Task :qa:restart-upgrade:delombok NO-SOURCE
> Task :qa:restart-upgrade:javadoc NO-SOURCE
> Task :qa:restart-upgrade:javadocJar UP-TO-DATE
> Task :qa:restart-upgrade:sourcesJar UP-TO-DATE
> Task :qa:restart-upgrade:assemble SKIPPED
> Task :qa:restart-upgrade:generateTestEffectiveLombokConfig UP-TO-DATE
> Task :qa:restart-upgrade:compileTestJava UP-TO-DATE
> Task :qa:restart-upgrade:processTestResources UP-TO-DATE
> Task :qa:restart-upgrade:testClasses UP-TO-DATE
> Task :qa:restart-upgrade:forbiddenApisResources UP-TO-DATE
> Task :qa:restart-upgrade:forbiddenApisMain NO-SOURCE
> Task :qa:restart-upgrade:forbiddenApisTest
> Task :qa:restart-upgrade:forbiddenApis
> Task :qa:restart-upgrade:dependencyLicenses SKIPPED
> Task :qa:restart-upgrade:filepermissions UP-TO-DATE
> Task :qa:restart-upgrade:forbiddenPatterns SKIPPED
> Task :qa:restart-upgrade:jarHell
> Task :qa:restart-upgrade:licenseHeaders UP-TO-DATE
> Task :qa:restart-upgrade:loggerUsageCheck SKIPPED
> Task :qa:restart-upgrade:testingConventions SKIPPED
> Task :qa:restart-upgrade:thirdPartyAuditResources UP-TO-DATE
> Task :qa:restart-upgrade:thirdPartyAudit SKIPPED
> Task :qa:restart-upgrade:validateNebulaPom SKIPPED
> Task :qa:restart-upgrade:validatePom UP-TO-DATE
> Task :qa:restart-upgrade:precommit
> Task :qa:restart-upgrade:integTest SKIPPED
> Task :qa:restart-upgrade:spotlessJava
> Task :qa:restart-upgrade:spotlessJavaCheck
> Task :qa:restart-upgrade:spotlessLicense UP-TO-DATE
> Task :qa:restart-upgrade:spotlessLicenseCheck UP-TO-DATE
> Task :qa:restart-upgrade:spotlessMisc UP-TO-DATE
> Task :qa:restart-upgrade:spotlessMiscCheck UP-TO-DATE
> Task :qa:restart-upgrade:spotlessCheck
> Task :qa:restart-upgrade:test SKIPPED
> Task :qa:restart-upgrade:check
> Task :qa:restart-upgrade:build SKIPPED
> Task :qa:rolling-upgrade:generatePomFileForNebulaPublication
> Task :qa:rolling-upgrade:generatePom
> Task :qa:rolling-upgrade:generateEffectiveLombokConfig UP-TO-DATE
> Task :qa:rolling-upgrade:compileJava NO-SOURCE
> Task :qa:rolling-upgrade:processResources NO-SOURCE
> Task :qa:rolling-upgrade:classes UP-TO-DATE
> Task :qa:rolling-upgrade:jar UP-TO-DATE
> Task :qa:rolling-upgrade:delombok NO-SOURCE
> Task :qa:rolling-upgrade:javadoc NO-SOURCE
> Task :qa:rolling-upgrade:javadocJar UP-TO-DATE
> Task :qa:rolling-upgrade:sourcesJar UP-TO-DATE
> Task :qa:rolling-upgrade:assemble SKIPPED
> Task :qa:rolling-upgrade:generateTestEffectiveLombokConfig UP-TO-DATE
> Task :qa:rolling-upgrade:compileTestJava UP-TO-DATE
> Task :qa:rolling-upgrade:processTestResources UP-TO-DATE
> Task :qa:rolling-upgrade:testClasses UP-TO-DATE
> Task :qa:rolling-upgrade:forbiddenApisResources UP-TO-DATE
> Task :qa:rolling-upgrade:forbiddenApisMain NO-SOURCE
> Task :qa:rolling-upgrade:forbiddenApisTest
> Task :qa:rolling-upgrade:forbiddenApis
> Task :qa:rolling-upgrade:dependencyLicenses SKIPPED
> Task :qa:rolling-upgrade:filepermissions UP-TO-DATE
> Task :qa:rolling-upgrade:forbiddenPatterns SKIPPED
> Task :qa:rolling-upgrade:jarHell
> Task :qa:rolling-upgrade:licenseHeaders UP-TO-DATE
> Task :qa:rolling-upgrade:loggerUsageCheck SKIPPED
> Task :qa:rolling-upgrade:testingConventions SKIPPED
> Task :qa:rolling-upgrade:thirdPartyAuditResources UP-TO-DATE
> Task :qa:rolling-upgrade:thirdPartyAudit SKIPPED
> Task :qa:rolling-upgrade:validateNebulaPom SKIPPED
> Task :qa:rolling-upgrade:validatePom UP-TO-DATE
> Task :qa:rolling-upgrade:precommit
> Task :qa:rolling-upgrade:integTest SKIPPED
> Task :qa:rolling-upgrade:spotlessJava
> Task :qa:rolling-upgrade:spotlessJavaCheck
> Task :qa:rolling-upgrade:spotlessLicense UP-TO-DATE
> Task :qa:rolling-upgrade:spotlessLicenseCheck UP-TO-DATE
> Task :qa:rolling-upgrade:spotlessMisc UP-TO-DATE
> Task :qa:rolling-upgrade:spotlessMiscCheck UP-TO-DATE
> Task :qa:rolling-upgrade:spotlessCheck
> Task :qa:rolling-upgrade:test SKIPPED
> Task :qa:rolling-upgrade:check
> Task :qa:rolling-upgrade:build SKIPPED

```